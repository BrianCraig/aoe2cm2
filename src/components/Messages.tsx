import * as React from 'react';
import '../pure-min.css'
import '../style2.css'
import {Trans, WithTranslation, withTranslation} from "react-i18next";
import Countdown from "../containers/Countdown";
import Player from "../models/Player";
import Turn from "../models/Turn";
import Action from "../models/Action";

interface IProps extends WithTranslation {
    whoAmI: Player | undefined,
    hostReady: boolean,
    guestReady: boolean,
    nextTurn: Turn | null;
    message: string
}

class Messages extends React.Component<IProps, object> {
    public render() {

        if (!this.props.hostReady || !this.props.guestReady) {
            if (this.props.whoAmI === Player.HOST) {
                if (this.props.hostReady) {
                    return (
                        <div><Trans>Waiting for Guest to press ›ready‹</Trans></div>
                    );
                } else {
                    if (this.props.guestReady) {
                        return (
                            <div><Trans>Your guest is ready to start. Press [ready] once you are also ready to
                                start!</Trans></div>
                        );
                    } else {
                        return (
                            <div><Trans>Press [ready] once you are ready to start!</Trans></div>
                        );
                    }
                }
            } else {
                if (this.props.guestReady) {
                    return (
                        <div><Trans>Waiting for Host to press ›ready‹</Trans></div>
                    );
                } else {
                    if (this.props.hostReady) {
                        return (
                            <div><Trans>Your host is ready to start. Press [ready] once you are also ready to
                                start!</Trans></div>
                        );
                    } else {
                        return (
                            <div><Trans>Press [ready] once you are ready to start!</Trans></div>
                        );
                    }
                }

            }
        }

        const nextTurn = this.props.nextTurn;
        if (nextTurn !== null) {
            if (nextTurn.player === this.props.whoAmI) {
                switch (nextTurn.action) {
                    case Action.PICK:
                        return (
                            <div><Trans><span className='green-glow'><b>Pick</b></span> a civilization!</Trans>
                                <Countdown/></div>
                        );
                    case Action.BAN:
                        return (
                            <div><Trans><span className='red-glow'><b>Ban</b></span> a civilization!</Trans>
                                <Countdown/></div>
                        );
                    case Action.SNIPE:
                        return (
                            <div><Trans><span className='yellow-glow'><b>Snipe</b></span> a civilization of the
                                opponent!</Trans> <Countdown/></div>
                        );
                }
            } else if (nextTurn.player === Player.NONE) {
                const action = nextTurn.action.toString();
                return (
                    <div><Trans>Admin action: {action}</Trans></div>
                );
            } else {
                switch (nextTurn.action) {
                    case Action.PICK:
                        return (
                            <div><Trans>Waiting for the other captain to pick…</Trans> <Countdown/></div>
                        );
                    case Action.BAN:
                        return (
                            <div><Trans>Waiting for the other captain to ban…</Trans> <Countdown/></div>
                        );
                    case Action.SNIPE:
                        return (
                            <div><Trans>Waiting for the other captain to snipe one of your civilisations…</Trans>
                                <Countdown/></div>
                        );
                }
            }


            const message = nextTurn.player.toString() + ': ' + nextTurn.action.toString();
            return (
                <div><Trans>{message}</Trans></div>
            );
        }

        return (
            <div><Trans>{this.props.message}</Trans> <Countdown/></div>
        );
    }
}

export default withTranslation()(Messages);
