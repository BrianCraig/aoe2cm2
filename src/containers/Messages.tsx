import {IStoreState} from '../types';
import {connect} from 'react-redux';
import Messages from "../components/Messages";
import {Dispatch} from "react";
import * as actions from "../actions";


export function mapStateToProps(state: IStoreState) {
    let nextTurn = null;
    if (state.preset && state.nextAction < state.preset.turns.length) {
        nextTurn = state.preset.turns[state.nextAction];
    }
    return {
        whoAmI: state.whoAmI,
        hostReady: state.hostReady,
        guestReady: state.guestReady,
        nameHost: state.nameHost,
        nameGuest: state.nameGuest,
        nextTurn
    };
}

export function mapDispatchToProps(dispatch: Dispatch<actions.Action>) {
    return {
        sendReady: () => dispatch(actions.sendReady())
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Messages);
