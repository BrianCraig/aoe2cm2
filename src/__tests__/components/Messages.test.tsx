import {shallow} from "enzyme";
import * as React from "react";
import Messages from "../../components/Messages";


it('Messages renders correctly', () => {
    const component = shallow(<Messages message={'message content'}/>);
    expect(component).toMatchSnapshot();
});
