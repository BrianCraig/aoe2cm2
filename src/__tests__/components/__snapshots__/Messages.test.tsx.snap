// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages renders correctly 1`] = `
<Messages
  i18n={Object {}}
  message="message content"
  t={[Function]}
  tReady={true}
/>
`;
